### PR TITLE
Moving Ion Mobility Ansgar Korf work

### PR DESCRIPTION
**Pull request by knakul853**
There is already some work is done by @Annexhc  regarding ion mobility in mzMine2 here https://github.com/Annexhc/mzmine2/tree/IonMobility but there is a lot of change have been made so far so in this PR I will include his work. Then as we are ok with the data support we move towards the visualizations.

1. Data support

- [x] moving basic structure/testing
- [x] resolving dependencies conflicts/testing 
- [x] fixing style issue and adding necessary docstring/testing
- [x] ask for review/testing
- [ ] solve suggested bugs by mentor/testing

2. Visualization
example

- [ ] presenting a basic structure of visualization to mentors.
- [ ] discussing technological details
- [ ] raw implementations
- [ ] fixing issue with raw implementations
- [ ] improving the implementation
- [ ] fixing bugs and asking for a review

### CODE REVIEW

**Comment by Annexhc** [comment]
Thanks Nakul, I will review it on Saturday!

**Comment by knakul853** [comment]
@Annexhc  Hey due to changes from swing to JavaFX all methods in main windows https://github.com/mzmine/mzmine3/tree/master/src/main/java/io/github/mzmine/gui/mainwindow 
your implementation of the  https://github.com/Annexhc/mzmine2/blob/IonMobility/src/main/java/net/sf/mzmine/desktop/impl/MainWindow.java#L332 these methods will take some time. Will soon ask for review : )

**Comment by Annexhc** [comment]
Hey Nakul, we should discuss this first! Just because I changed the GUI back then does not mean it is the rigth thing to do! Let's discuss this tomorrow, I will go to bed now.

**Comment by knakul853** [comment]
Ok.btw I keep on testing till now all worked fine :)

**Comment by Annexhc** [comment]
Hey Nakul, after some testing, it looks quite good (see screenshot).
Anyway, we should go a step back first and focus more on planning/organizing. A lot of changes you did in this PR will be not very useful in the future, because we are about to change the feature data model. Of course you couldn't know that, but that's why we should have some more discussions in the future :-). We should move those to slack for the time of GSoC. 

Coming back to this PR:
We should break this into smaller pieces. We should focus on the raw data support itself first and postpone all changes to the GUI for now (e.g. we don't need to think about "MobilogramIdentity" at the moment). Therefore, the important classes/interfaces you changed are:

RawDataFile
RawDataFileImpl
Scan
SimpleScan
StorableScan
MzMLReadTask

All changes related to "MobilogramList" are not important at the moment, because with the new feature data model (which is work in progress), we will get much more options to implement visualizations of picked IMS Features.
Therefore, I suggest to close this PR and we will focus together with Aivett on your IMS development branch. We can than make a final PR to the master branch of MZmine 3 once you project is finished. 

image

**Comment by knakul853** [comment]
closing this PR as suggested by @Annexhc will working on my branch and then reopen this PR

**Comment by knakul853** [comment]
Re-opening this PR as we discussed

File: src/main/java/io/github/mzmine/MobilogramListRow.java [hunk 1]
```diff
@@ -0,0 +1,200 @@
+package io.github.mzmine;
```
Comment (Annexhc) [inline src/main/java/io/github/mzmine/MobilogramListRow.java:1]: Remove this Interface

File: src/main/java/io/github/mzmine/datamodel/IMSDataPoint.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package io.github.mzmine.datamodel;
```
Comment (Annexhc) [inline src/main/java/io/github/mzmine/datamodel/IMSDataPoint.java:1]: Remove this Interface

File: src/main/java/io/github/mzmine/datamodel/IMSFeature.java [hunk 3]
```diff
@@ -0,0 +1,172 @@
+package io.github.mzmine.datamodel;
```
Comment (Annexhc) [inline src/main/java/io/github/mzmine/datamodel/IMSFeature.java:1]: Remove this Interface

File: src/main/java/io/github/mzmine/datamodel/MobilogramIdentity.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
+package io.github.mzmine.datamodel;
```
Comment (Annexhc) [inline src/main/java/io/github/mzmine/datamodel/MobilogramIdentity.java:1]: Remove this Interface

File: src/main/java/io/github/mzmine/datamodel/MobilogramInformation.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
+package io.github.mzmine.datamodel;
```
Comment (Annexhc) [inline src/main/java/io/github/mzmine/datamodel/MobilogramInformation.java:1]: Remove this Interface

File: src/main/java/io/github/mzmine/datamodel/MobilogramList.java [hunk 6]
```diff
@@ -0,0 +1,206 @@
+package io.github.mzmine.datamodel;
```
Comment (Annexhc) [inline src/main/java/io/github/mzmine/datamodel/MobilogramList.java:1]: Remove this Interface

File: src/main/java/io/github/mzmine/datamodel/MobilogramListRow.java [hunk 7]
```diff
@@ -0,0 +1,197 @@
+package io.github.mzmine.datamodel;
```
Comment (Annexhc) [inline src/main/java/io/github/mzmine/datamodel/MobilogramListRow.java:1]: Remove this Interface

File: src/main/java/io/github/mzmine/datamodel/impl/SimpleMobilogramInformation.java [hunk 8]
```diff
@@ -0,0 +1,58 @@
+package io.github.mzmine.datamodel.impl;
```
Comment (Annexhc) [inline src/main/java/io/github/mzmine/datamodel/impl/SimpleMobilogramInformation.java:1]: Remove this class

File: src/main/java/io/github/mzmine/datamodel/impl/SimpleScan.java [hunk 9]
```diff
@@ -132,10 +134,6 @@ public SimpleScan(RawDataFile dataFile, int scanNumber, int msLevel, double rete
     return pointsOverIntensity;
   }
 
-  /**
```
Comment (Annexhc) [inline src/main/java/io/github/mzmine/datamodel/impl/SimpleScan.java:135]: Why did you remove this?

File: src/main/java/io/github/mzmine/datamodel/impl/SimpleScan.java [hunk 10]
```diff
@@ -285,9 +296,6 @@ public MassSpectrumType getSpectrumType() {
     return spectrumType;
   }
 
-  /**
```
Comment (Annexhc) [inline src/main/java/io/github/mzmine/datamodel/impl/SimpleScan.java:288]: dont remove this

File: src/main/java/io/github/mzmine/gui/Desktop.java [hunk 11]
```diff
@@ -19,6 +19,7 @@
 package io.github.mzmine.gui;
```
Comment (Annexhc) [inline src/main/java/io/github/mzmine/gui/Desktop.java:19]: No changes needed here

File: src/main/java/io/github/mzmine/gui/HeadLessDesktop.java [hunk 12]
```diff
@@ -22,13 +22,13 @@
 import java.util.logging.Level;
```
Comment (Annexhc) [inline src/main/java/io/github/mzmine/gui/HeadLessDesktop.java:22]: No changes needed here

File: src/main/java/io/github/mzmine/project/impl/MZmineProjectImpl.java [hunk 13]
```diff
@@ -21,7 +21,6 @@
 import java.io.File;
```
Comment (Annexhc) [inline src/main/java/io/github/mzmine/project/impl/MZmineProjectImpl.java:21]: no changes needed here

File: src/main/java/io/github/mzmine/project/impl/RawDataFileImpl.java [hunk 14]
```diff
@@ -481,6 +481,12 @@ public synchronized RawDataFile finishWriting() throws IOException {
     return getDataRTRange(0);
   }
 
+  @Nonnull
```
Comment (Annexhc) [inline src/main/java/io/github/mzmine/project/impl/RawDataFileImpl.java:484]: This is more or less dummy code. For know it is okay I guess, since the raw data import works. You need a similar method like "getDataRTRange()" or "getDataMZRange() for the mobility range. We also have to think about the storing of the data. But I think we can do this in another pull request.

File: src/main/java/io/github/mzmine/datamodel/impl/SimpleScan.java (hunk 9 above)
Comment (knakul853) [reply src/main/java/io/github/mzmine/datamodel/impl/SimpleScan.java:135]: I think by mistake done this.added

**Comment by knakul853** [comment]
Extraction of mobility us working fine, but in rawdataoverview  i add a new column it reflects the value as "0". will see this tomorrow :)

File: src/main/java/io/github/mzmine/datamodel/IMSDataPoint.java (hunk 2 above)
Comment (tomas-pluskal) [inline src/main/java/io/github/mzmine/datamodel/IMSDataPoint.java:1]: If this interface is not used, it should be removed

**Comment by tomas-pluskal** [comment]
I think we can merge this, but please remove the IMSDataPoint interface

**Comment by tomas-pluskal** [comment]
Thanks!
